ledger/src/blockstore_processor.rs

```diff
@@ -941,83 +937,75 @@ pub struct ConfirmationTiming {
+        }
+    }
+}
+
+/// Measures times related to transaction execution in a slot.
+#[derive(Debug, Default)]
+pub struct BatchExecutionTiming {
+    /// Time used by transaction execution.  Accumulated across multiple threads that are running
+    /// `execute_batch()`.
+    pub totals: ExecuteTimings,
+
+    /// Wall clock time used by the transaction execution part of pipeline.
+    /// [`ConfirmationTiming::replay_elapsed`] includes this time.  In microseconds.
+    pub wall_clock_us: u64,
 
     /// Time used to execute transactions, via `execute_batch()`, in the thread that consumed the
     /// most time.
-    pub end_to_end_execute_timings: ThreadExecuteTimings,
+    pub slowest_thread: ThreadExecuteTimings,
 }
 
-impl ConfirmationTiming {
-    fn process_execute_batches_internal_metrics(
-        &mut self,
-        execute_batches_internal_metrics: ExecuteBatchesInternalMetrics,
-    ) {
-        let ConfirmationTiming {
-            execute_timings: cumulative_execute_timings,
-            execute_batches_us: cumulative_execute_batches_us,
-            end_to_end_execute_timings,
-            ..
+impl BatchExecutionTiming {
+    fn accumulate(&mut self, new_batch: ExecuteBatchesInternalMetrics) {
+        let Self {
+            totals,
+            wall_clock_us,
+            slowest_thread,
         } = self;
 
-        saturating_add_assign!(
-            *cumulative_execute_batches_us,
-            execute_batches_internal_metrics.execute_batches_us
-        );
+        saturating_add_assign!(*wall_clock_us, new_batch.execute_batches_us);
 
-        cumulative_execute_timings.saturating_add_in_place(
-            ExecuteTimingType::TotalBatchesLen,
-            execute_batches_internal_metrics.total_batches_len,
-        );
-        cumulative_execute_timings.saturating_add_in_place(ExecuteTimingType::NumExecuteBatches, 1);
+        use ExecuteTimingType::{NumExecuteBatches, TotalBatchesLen};
+        totals.saturating_add_in_place(TotalBatchesLen, new_batch.total_batches_len);
+        totals.saturating_add_in_place(NumExecuteBatches, 1);
 
-        let mut current_max_thread_execution_time: Option<ThreadExecuteTimings> = None;
-        for (_, thread_execution_time) in execute_batches_internal_metrics
-            .execution_timings_per_thread
-            .into_iter()
-        {
-            let ThreadExecuteTimings {
-                total_thread_us,
-                execute_timings,
-                ..
-            } = &thread_execution_time;
-            cumulative_execute_timings.accumulate(execute_timings);
-            if *total_thread_us
-                > current_max_thread_execution_time
-                    .as_ref()
-                    .map(|thread_execution_time| thread_execution_time.total_thread_us)
-                    .unwrap_or(0)
-            {
-                current_max_thread_execution_time = Some(thread_execution_time);
+        let mut slowest = None;
+        let mut slowest_time = 0;
```

Comment:
nit: would like to distinguish better between "slowest" and "slowest_time", sound a bit too similar.  

Maybe

"slowest" -> "slowest_thread_details"
"slowest_time" -> "slowest_total_us"

Reply:
I wanted to keep them alike, as they both reference the same thing.
Just remembered that `Iterator` provides `max_by_key()`, so there is no need to write the search explicitly.
This removes `slowest_time` :)
And also makes the function even shorter.

What do you think?

Reply:
seems much cleaner to me!